### ISSUE TITLE
Caret Freezing

### ISSUE THREAD

**Issue by langsor** [thomaswilburn/Caret#723]
### Type of issue (choose one):
  - [X] Bug

### Test steps

Recently, when I type into Caret it simply freezes. 

This started happening within the last couple days.

This happens repeatedly every time I open and try to use it. I reset it's prefs. I reinstalled it. I shut down all other apps. No go.

I can not save work, I can not shut it down. I have to "close" it from the launch bar.

### Description

To not freeze.

### System details

ChromeOS, latest version, developer mode.

Asus Chromebook

![image](https://user-images.githubusercontent.com/5402040/77275500-af862900-6c75-11ea-822b-aeab5b6facd7.png)

**Comment by leesajay** [comment]
I have also experienced this, three times in the last week or so. I have not (yet) reset or uninstalled. Quitting and reopening allows me to continue using the app. I am on the latest Chrome OS not in developer mode. I will turn on error collection mode and report back.

**Comment by MaseratiPilot** [comment]
I am also having this same problem.
I have to close the app.

One time I waited 20 minutes or so (didn't time it exactly) and it came back.  Within a few minutes it happened again.

Any ideas?

**Comment by thomaswilburn** [comment]
On this end, nothing has substantially changed in Caret's code for at least a year. I don't know why this is happening. But given Google's plan to deprecate these apps very soon, I don't have a lot of time to try debugging it. I will happily take a pull request if someone provides a fix, however.

**Comment by langsor** [comment]
Bummer. Yet, I understand.
I rely heavily on Caret for development and everything else on my Chromebook.
I've never bought any upgrade package because the free version has everything I need. But such a good app deserves to stay alive.
When you say Google is deprecating these apps, is that the decision they made in 2018? Could Caret be ported as a Progressive Web App and still be as robust as it is now? Most PWAs are pretty clunky still.
My guess is that some ChromeOS update broke Caret. Sadly this is out of my ability to troubleshoot.
The article I read says that Google intends to keep apps alive for Chromebooks. Not sure what caveats that comes with.
Let me know if there's anything I can do to support making Caret work again. 
Not sure how to access the source code / internals of Caret. 
I'm still having this problem and having to look for alternatives--there isn't anything else good out there.

Either way, thank you for making such a good app available. It's been really good.

Cheers,
Kevin

**Comment by leohomer** [comment]
I want to let you know this happened to me as well.  I had to kill the window each time.  I tried restarting the background page, but that didn't help.  I happened upon a version of cloud 9 that also uses the ace editor that run with local storage.  I tried using it for a while and then the window it was in locked up the same way.  I couldn't move it, use any keys, even the mouse wasn't working.  I wondering if this a chrome problem with ace and not caret.

**Comment by leesajay** [comment]
This may or may not be a helpful update but I have not had any more problems since the latest Chrome OS update (last week?).

**Comment by leohomer** [comment]
Sorry. No dice.  I got the update too.  I also tried a third program that
uses ace on chrome.  It was an older program in the web store, that is no
longer supported and hadn't been update since 2018.  I thought maybe it was
using an older version of ace and that would make the difference.  Nope,
that one locked almost as soon as I started using it.  This may be a
complication of ace under chrome.  So your program, the other editor code
pad at
https://chrome.google.com/webstore/detail/code-pad-text-editor/adaepfiocmagdimjecpifghcgfjlfmkh
and
a cloud 9 online using local storage, but using ace at
https://c9.github.io/core/ just to try it.  All three locked up.  Yesterday
I file I had just finished coding and was looking at in Caret when it
locked got zeroed out and I lost all the changes.  Luckily I had backed up
the original, but I can't take that chance anymore.

Thanks,
Mark
